core/src/main/scala/com/microsoft/azure/synapse/ml/causal/DoubleMLEstimator.scala

```diff
@@ -246,11 +267,16 @@ class DoubleMLEstimator(override val uid: String)
       .setLink("identity")
       .setFitIntercept(false)
 
-    val coefficients = Array(residualsDF1, residualsDF2).map(regressor.fit).map(_.coefficients(0))
-    val ate = coefficients.sum / coefficients.length
+    val regressorModel1 = regressor.fit(residualsDF1)
+    val regressorModel2 = regressor.fit(residualsDF2)
+    val ate = (regressorModel1.coefficients(0) + regressorModel2.coefficients(0)) / 2.0
 
     Seq(train, test).foreach(_.unpersist)
-    ate
+    (ate,
+      Array(treatmentModel1, treatmentModel2),
+      Array(outcomeModel1, outcomeModel2),
+      Array(regressorModel1, regressorModel2)
+    )
```

Comment:
It may be better to reshape the output here to Tuple of Array:
```suggestion
    (ate,
      (Array(treatmentModel1, outcomeModel1, regressorModel1), Array(treatmentModel2, outcomeModel2, regressorModel2))
    )
```
Or even better, put the treatmentModel, outcomeModel, regressorModel triplets into a pipeline model's stages, and just return the 2 pipeline models.